### ISSUE TITLE
[gokz-core] Couldn't find gamerules proxy entity

### ISSUE THREAD

**Issue by walliski** [KZGlobalTeam/gokz#309]
Getting this on mapchange. What I have understood it happens because the gamerules are trying to be accessed before they are properly loaded. What I know of, you cannot really know if they are loaded or not though...

3.2.1

```
L 04/04/2022 - 21:08:06: SourceMod error session started
L 04/04/2022 - 21:08:06: Info (map "kz_antigeneric") (file ".../addons/sourcemod/logs/errors_20220404.log")
L 04/04/2022 - 21:08:06: [SM] Exception reported: Couldn't find gamerules proxy entity
L 04/04/2022 - 21:08:06: [SM] Blaming: gokz-core.smx
L 04/04/2022 - 21:08:06: [SM] Call stack trace:
L 04/04/2022 - 21:08:06: [SM]   [0] GameRules_SetProp
L 04/04/2022 - 21:08:06: [SM]   [1] Line 75, gokz-core/demofix.sp::DoDemoFix
L 04/04/2022 - 21:08:06: [SM]   [2] Line 55, gokz-core/demofix.sp::OnDemofixConVarChanged
L 04/04/2022 - 21:08:06: [SM] Exception reported: Couldn't find gamerules proxy entity
L 04/04/2022 - 21:08:06: [SM] Blaming: gokz-core.smx
L 04/04/2022 - 21:08:06: [SM] Call stack trace:
L 04/04/2022 - 21:08:06: [SM]   [0] GameRules_SetProp
L 04/04/2022 - 21:08:06: [SM]   [1] Line 75, gokz-core/demofix.sp::DoDemoFix
L 04/04/2022 - 21:08:06: [SM]   [2] Line 55, gokz-core/demofix.sp::OnDemofixConVarChanged
```

**Comment by sneak-it** [comment]
Solution here would be to set a bool OnMapStart to true, and false at OnMapEnd. if (bool == true) do demo fix.

**Comment by walliski** [comment]
> Solution here would be to set a bool OnMapStart to true, and false at OnMapEnd. if (bool == true) do demo fix.

Apparently that was already done, but not in the case of having the DemoFix set to "Disabled".

**Comment by walliski** [comment]
Fixed in #318
